pkg/ingestr/operator.go

```diff
@@ -0,0 +1,74 @@
+package ingestr
+
+import (
+	"context"
+	"errors"
+
+	"github.com/bruin-data/bruin/pkg/executor"
+	"github.com/bruin-data/bruin/pkg/pipeline"
+	"github.com/bruin-data/bruin/pkg/scheduler"
+)
+
+type BasicOperator struct{}
+
+func NewBasicOperator() *BasicOperator {
+	return &BasicOperator{}
+}
+
+func (o BasicOperator) Run(ctx context.Context, ti scheduler.TaskInstance) error {
+	return o.RunTask(ctx, ti.GetPipeline(), ti.GetAsset())
+}
+
+func (o BasicOperator) RunTask(ctx context.Context, p *pipeline.Pipeline, t *pipeline.Asset) error {
+	return errors.New("Ingestr tasks not implemented")
+}
+
+type IngestrCheckOperator struct {
+	configs *map[pipeline.AssetType]executor.Config
```

Comment:
why is the value here of type `executor.Config` but not an actual column check operator instance?

Reply:
The idea is that the ingestr check operator needs to be able to perform checks for different types of destination. To illustrate with an example, imagine you have a pipeline where you have 2 ingestr assets, one has a postgres destination and another one has a mssql destination. The are both ingestr so they would use this check operator but the checks need to be different